kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiBatchSession.scala

```diff
@@ -82,8 +82,8 @@ class KyuubiBatchSession(
   override val normalizedConf: Map[String, String] =
     sessionConf.getBatchConf(batchType) ++ sessionManager.validateBatchConf(conf)
 
-  private[kyuubi] def resourceUploadFolderPath: Path =
-    BatchesResource.batchResourceUploadFolderPath(batchJobSubmissionOp.batchId)
+  private[kyuubi] lazy val resourceUploadFolderPath: Path =
```

Comment:
```suggestion
  private[kyuubi] def resourceUploadFolderPath: Path =
```
Minor optional suggestion: function instead of holding the variable, as it's able to be recalculated.